builder/qemu/step_wait_guest_address.go

```diff
@@ -0,0 +1,122 @@
+package qemu
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+
+	"github.com/digitalocean/go-qemu/qmp"
+)
+
+// This step waits for the guest address to become available in the network
+// bridge, then it sets the guestAddress state property.
+type stepWaitGuestAddress struct{}
+
+func (s *stepWaitGuestAddress) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	config := state.Get("config").(*Config)
+	ui := state.Get("ui").(packer.Ui)
+	qmpMonitor := state.Get("qmp_monitor").(*qmp.SocketMonitor)
+
+	ui.Say(fmt.Sprintf("Waiting for the guest address to become available in the %s network bridge...", config.NetBridge))
+	for {
```

Comment:
Should this step have a timeout so that it doesn't loop forever if something is wrong?

Reply:
Should it be hard-coded to value (e.g. 5m)? Or should it introduce another builder config variable? Or is there some variable that I could re-use (e.g. ssh_timeout/winrm_timeout)?

Reply:
sorry I forgot to answer this. I think for now we can use the ssh_timeout and if that doesn't make sense in the future we can change.

Reply:
Do you want this in another PR?

Reply:
Oh, oops. Forgot I was waiting on this. Yes, another PR would be awesome.

Reply:
Its at https://github.com/hashicorp/packer/pull/9287